lumigator/frontend/Dockerfile

```diff
@@ -7,6 +7,13 @@ COPY ../../ /mzai
 
 WORKDIR /mzai/
 
+# Copy the .env file for the entire project, and perform some transformation on it.
+# Only env vars starting with the 'VUE_APP_' prefix are made visible by the Vite build system (See: vite.config.js),
+# but we extract only these variables in a kind of 'least privilege' way.
+# See: https://cli.vuejs.org/guide/mode-and-env.html#environment-variables
+COPY .env /mzai/lumigator/frontend/
+RUN cat .env | grep -E ^VUE_APP_ > /mzai/lumigator/frontend/.env
```

Comment:
If we are interested only in `VUE_APP_*` vars, do we need an `.env` file copied into frontend (and to allow .env files at all in dockerignore)?
Unless it is required locally too, we might as well have a `.vue_environment` (just making up a name here) file and then `COPY .vue_environment /mzai/lumigator/frontend/.env`. WDYT?

(still wondering why Vite cannot access env vars from the environment but looks for them in a file, but I think the simplest solution would work for now while we think about how to implement this in a way which is more consistent with the rest of the stack)

Reply:
I believe the intention behind our single `.env` file was to consolidate the env vars, having a different file like the example you gave (`.vue_environment`) would mean we're back to having env vars in different places. 🤔 

Vite (the build system) can access actual environment variables during the build (they take priority over [`.env`](https://github.com/motdotla/dotenv) ones), but env vars supplied via docker compose only appear in the finished frontend container. The build tool doesn't have access to them, as they would need to exist in the first stage of the multi-stage build we do (that we throw away). 

From: https://vite.dev/guide/env-and-mode#env-files

> In addition, environment variables that already exist when Vite is executed have the highest priority and will not be overwritten by `.env` files.

This solution does ensure that we only need to configure the variable in a single place, and the Dockerfile only ends up with `VUE_APP_` prefixed vars which are part of the throw-away stage.  

I'm probably missing something but I don't fully understand why the proposed solution might not be acceptable (even as a short term one). 😢

Reply:
I'm going to have a play with 'one more thing' to see if there's a bit of a workaround for the docker compose file...

Reply:
Victory? 581a6742519114d91a6aeee0912885bce219e32b

Reply:
@aittalam Please remember that this setting needs to be set up at image build time, so it won't be reloaded by the final container if it is changed. The final container has only static files that cannot be changed at the time the UI starts. 
@peteski22 approach seems fine, and can use env vars already present in your environment, so I'd approve this PR now, and inspect the build system later. Wdyt?